### PR TITLE
View replies 843

### PR DESCRIPTION
**Pull request by yulkats**
1. EcoNews comments have been rewritten;
2. Replies to comments have been rewritten;
[843](https://github.com/ita-social-projects/GreenCity/issues/843)

BugFix:

- Count of replies (after adding/deleting) changes after updating 'eco news' page [976](https://github.com/ita-social-projects/GreenCity/issues/976);

- Count after deleting comments works after updating 'eco news' page [967](https://github.com/ita-social-projects/GreenCity/issues/967);

- Error message pop-up is opened after clicking on 'Edit' button on 'Eco news' Single Page. [1401](https://github.com/ita-social-projects/GreenCity/issues/1401);

- Comment text is disappeared after clicking on 'Edit' button on 'Eco news' Single Page. [1242](https://github.com/ita-social-projects/GreenCity/issues/1242);

- The system recalculates the total likes number only after refreshing the page. [1057](https://github.com/ita-social-projects/GreenCity/issues/1057);

### CODE REVIEW

File: src/app/component/comments/components/comments-container/comments-container.component.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        .subscribe((list: CommentsModel) => this.setCommentsList(list));
+    }
+  }
+
+  public setCommentsList(data: CommentsModel): void {
+    this.elementsList = [...data.page];
+    this.elementsArePresent = this.elementsList.length > 0;
+  }
+
+  private checkUserSingIn(): void {
```
Comment (mr-pavlenko) [inline src/app/component/comments/components/comments-container/comments-container.component.ts:67]: don't mix public and private

File: src/app/component/comments/components/reply-comment/reply-comment.component.ts [hunk 2]
```diff
@@ -6,13 +6,10 @@ import { Component, OnInit } from '@angular/core';
...
 })
 export class ReplyCommentComponent implements OnInit {
-  public commentsImages = {
-    reply: 'assets/img/comments/reply.png'
-  };
+  public replyImages = 'assets/img/comments/reply.png';
 
   constructor() { }
 
   ngOnInit() {
```
Comment (mr-pavlenko) [inline src/app/component/comments/components/reply-comment/reply-comment.component.ts:13]: Why do you need ngOnInit ?

File: src/app/component/comments/components/like-comment/like-comment.component.ts [hunk 3]
```diff
@@ -30,15 +30,22 @@ export class LikeCommentComponent implements OnInit {
...
   public changeLkeBtn(): void {
     const cond = this.like.nativeElement.srcset === this.commentsImages.like;
     const imgName = cond ? 'liked' : 'like';
     this.like.nativeElement.srcset = this.commentsImages[imgName];
+    this.likeState = !this.likeState;
+  }
+
+  private getLikesFromServer(): void {
+    this.commentsService.getCommentLikes(this.comment.id)
+      .subscribe((data: number) => this.likesCounter.emit(data));
```
Comment (mr-pavlenko) [inline src/app/component/comments/components/like-comment/like-comment.component.ts:49]: What will happen in case error?

**Comment by mr-pavlenko** [comment]
Add screenshots

File: src/app/component/comments/components/comments-container/comments-container.component.ts (hunk 1 above)
Comment (yulkats) [reply src/app/component/comments/components/comments-container/comments-container.component.ts]: fixed

File: src/app/component/comments/components/reply-comment/reply-comment.component.ts (hunk 2 above)
Comment (yulkats) [reply src/app/component/comments/components/reply-comment/reply-comment.component.ts]: fixed

File: src/app/component/comments/components/like-comment/like-comment.component.ts (hunk 3 above)
Comment (yulkats) [reply src/app/component/comments/components/like-comment/like-comment.component.ts]: fixed

**Pull request merged**